### PR TITLE
Return list of existing shadowbox servers (to support auto-advancing on website)

### PR DESCRIPTION
**Pull request by dborkan**
This pull request:
* Adds a new method, `getExistingServers`, which returns info (invite, name, region) for any existing shadowbox servers
* Passes the server name (URI encoded) to the shadowbox install script, to be used in the manager's environment variables.

Along with this pull request, we need to:
1. Change the uProxy-website to auto-advance to step 3 (display invite) if the user already has a shadowsocks server
2. Change the shadowbox manager to read the SS_SERVER_NAME and append it to all invites (so Shadowsocks clients will show a name rather than IP address)

### CODE REVIEW

File: src/proxy-installers/install-shadowbox-on-digital-ocean.sh [hunk 1]
```diff
@@ -40,7 +43,7 @@ ufw disable
 mkdir -p $STATE_DIR
 
 # Start shadowbox docker container.
-docker run -d --name shadowbox --restart=always --net=host -v $STATE_DIR:$STATE_DIR -e "SS_PUBLIC_IP=$(dig +short myip.opendns.com @resolver1.opendns.com)" -e "SS_CONFIG_FILE=$CONFIG_FILE" fortuna/shadowbox
+docker run -d --name shadowbox --restart=always --net=host -v $STATE_DIR:$STATE_DIR -e "SS_PUBLIC_IP=$(dig +short myip.opendns.com @resolver1.opendns.com)" -e "SS_CONFIG_FILE=$CONFIG_FILE" -e "SS_SERVER_NAME=$SS_SERVER_NAME" jigsawcode/shadowbox
```
Comment (fortuna) [inline src/proxy-installers/install-shadowbox-on-digital-ocean.sh]: You don't need the `SS_PUBLIC_IP` variable anymore.

File: src/proxy-installers/shadowsocks-installer.ts [hunk 2]
```diff
@@ -21,17 +30,17 @@ export class ShadowsocksInstaller {
...
+      // Replacing SHADOWBOX_PREFIX with 'ss://' gives us the initial invite for
       // shadowsocks on that droplet.
       const INTERVAL_MS = 3000;
-      var digitalOceanSession = new DigitalOceanSession(digitalOceanAccessToken);
+      let digitalOceanSession = new DigitalOceanSession(digitalOceanAccessToken);
       let checkTags = () => {
         digitalOceanSession.getDropletTags(dropletId).then((tags: string[]) => {
-          for (var i = 0; i < tags.length; ++i) {
-            if (tags[i].indexOf('shadowbox-') === 0) {
+          for (let i = 0; i < tags.length; ++i) {
```
Comment (fortuna) [inline src/proxy-installers/shadowsocks-installer.ts]: Use `for (let tag of tags) {..}`

File: src/proxy-installers/shadowsocks-installer.ts [hunk 3]
```diff
@@ -45,3 +54,64 @@ export class ShadowsocksInstaller {
...
+      }
+    }
+    return shadowboxServers;
+  });
+}
+
+// Returns shadowbox invite if this droplet is for shadowbox, otherwise
+// returns null.
+function getShadowboxInvite(droplet: any): string {
+  for (let i = 0; i < droplet.tags.length; ++i) {
```
Comment (fortuna) [inline src/proxy-installers/shadowsocks-installer.ts]: `for (let tag of droplet.tags)`

**Comment by dborkan** [comment]
Just made those 3 changes, PTAL

**Pull request merged**